### PR TITLE
Migrate readme yaml

### PR DESCRIPTION
**Pull request by solairerove**
## what
- updated `README.yaml` file
- add tags and categories
- rebuild `README.md` file
- regenerate readme md
- add description key in outputs values
- add makefile
- ignore harness
- prepare to open discussions on review

## why
- need to add categories and tags so we can pull them into the documentation

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    image: "https://img.shields.io/github/release/cloudposse/terraform-aws-user-data-dns-proxy.svg"
+    url: "https://github.com/cloudposse/terraform-aws-user-data-dns-proxy/releases/latest"
+  - name: "Slack Community"
+    image: "https://slack.cloudposse.com/badge.svg"
+    url: "https://slack.cloudposse.com"
+
+related:
+  - name: "terraform-template-user-data-assests"
+    description: "terraform-template-user-data-assests"
+    url: "https://travis-ci.org/cloudposse/terraform-template-user-data-assests"
```
Comment (osterman) [inline README.yaml]: Wrong related URLs. These will never be travis.

File: docs/terraform.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+| name |  | string | - | yes |
+| namespace |  | string | - | yes |
+| region | AWS region instance running in | string | - | yes |
+| stage |  | string | - | yes |
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| user_data |  |
```
Comment (osterman) [inline docs/terraform.md]: Add description to the output. 
```
output "blah" {
  description = "...."
  value = "..."
}
```

Description: "Script to be executed by instance on startup"

File: docs/terraform.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| domain | Domain that should be resolved special way (point to var.ip address) | string | - | yes |
+| flavor | Flavor depends of OS and init system | string | `debian-systemd` | no |
+| ip | IP for domain | list | `<list>` | no |
+| name |  | string | - | yes |
```
Comment (osterman) [inline docs/terraform.md]: Get the name, namespace, stage descriptions from one of the other modules.

**Review by osterman — CHANGES_REQUESTED** [review]
See comments

File: README.yaml (hunk 1 above)
Comment (solairerove) [reply README.yaml]: done

File: docs/terraform.md (hunk 2 above)
Comment (solairerove) [reply docs/terraform.md]: done

File: docs/terraform.md (hunk 3 above)
Comment (solairerove) [reply docs/terraform.md]: done

**Comment by solairerove** [comment]
@osterman review plz

**Pull request closed without merging**